### PR TITLE
[Merged by Bors] - chore(Topology/Sheaves): split off results on sheaves of rings

### PR DESCRIPTION
**Pull request by Vierkantor**
I noticed when debugging #18734 that `Topology/Sheaves/Stalks.lean` imports ordered rings / groups / monoids, but those are not at all used. This will be fixed if we split off the import of `CommRingCat` in the sheaf constructions into their own files.

This moves all declarations in `Topology/Sheaves` involving `CommRingCat` into a new file `Sheaves/CommRingCat.lean`, except those of `Sheaves/Operations.lean` since that file is already entirely about sheaves of rings.

(I am not at all comfortable in this area of the library, so I expect the file name and docs are highly suboptimal. I welcome all corrections to my ignorance :D)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by mattrobball — COMMENTED** [review]
What other files in `Topology.Sheaves` import `CommRingCat`? Can we just call the new file `CommRingCat` and dump more things here?

**Comment by Vierkantor** [comment]
> My mental picture is import things from `Topology.Sheaves.CommRingCat` if you want sheaves valued in commutative rings. Otherwise, don't. Hopefully, we don't have to go through `CommRingCat` to get sheaves of abelian groups.

Ok! I'll try and do the change but I can't say for sure that I have the knowledge to do this cleanly.

**Review by Vierkantor — COMMENTED** [review]
Thank you for the changes and fixes :D

**Comment by mattrobball** [comment]
I merged `Topology.Sheaves.Operations` into `Topology.Sheaves.CommRingCat` also and `section`ed off with names corresponding to the source files.

File: Mathlib/Topology/Sheaves/CommRingCat.lean [hunk 1]
```diff
@@ -0,0 +1,329 @@
+/-
+Copyright (c) 2020 Kim Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison
```
Comment (mattrobball) [inline Mathlib/Topology/Sheaves/CommRingCat.lean:4]:
```suggestion
Authors: Kim Morrison, Andrew Yang
```

**Review by erdOne — APPROVED** [review]
LGTM
Thanks for doing this!

**Review by TwoFX — APPROVED** [review]
Thanks!
bors r+

**Pull request closed without merging**